src/weiroll/index.ts

```diff
@@ -0,0 +1,69 @@
+import { Planner } from '@weiroll/weiroll.js'
+import { Interface } from '@ethersproject/abi'
+import { EvmCall } from '../common'
+export type { Contract as WeirollContract } from '@weiroll/weiroll.js'
+
+export enum CommandFlags {
+  DELEGATECALL = 0,
+  CALL = 1,
+  STATICCALL = 2,
+  CALL_WITH_VALUE = 3,
+  CALLTYPE_MASK = 3,
+  EXTENDED_COMMAND = 64,
+  TUPLE_RETURN = 128,
+}
+
+const WEIROLL_ADDRESS = '0x9585c3062Df1C247d5E373Cfca9167F7dC2b5963'
+
+const WEIROLL_ABI = [
+  {
+    inputs: [
+      { internalType: 'uint256', name: 'command_index', type: 'uint256' },
+      { internalType: 'address', name: 'target', type: 'address' },
+      { internalType: 'string', name: 'message', type: 'string' },
+    ],
+    name: 'ExecutionFailed',
+    type: 'error',
+  },
+  {
+    inputs: [
+      { internalType: 'bytes32[]', name: 'commands', type: 'bytes32[]' },
+      { internalType: 'bytes[]', name: 'state', type: 'bytes[]' },
+    ],
+    name: 'execute',
+    outputs: [{ internalType: 'bytes[]', name: '', type: 'bytes[]' }],
+    stateMutability: 'payable',
+    type: 'function',
+  },
+]
+
+function getWeirollCalldata(planner: Planner) {
+  const planResult = planner.plan()
+  const { commands, state } = planResult
+
+  const weirollInterface = new Interface(WEIROLL_ABI)
+
+  return weirollInterface.encodeFunctionData('execute', [commands, state])
+}
+
+/**
+ * Create a weiroll transaction
+ *
+ * @param addToPlanner - A function that adds the commands to the planner
+ *
+ * @returns An EVM call
+ */
+export function createWeirollDelegateCall(addToPlanner: (planner: Planner) => void): EvmCall {
+  // Create a new planner
+  const planner = new Planner()
```

Comment:
Question: every time we create a new weiroll delegate call we must create a new planner instance? Can that ever be reused outside?

Given the usage pattern, it seems all operations needed will be done inside the `addToPlanner` callback, so I guess the answer is no.